tools/toolchains/__init__.py

```diff
@@ -201,6 +201,8 @@ def get_symbols(self, for_asm=False):
                 if MBED_ORG_USER:
                     self.cxx_symbols.append('MBED_USERNAME=' + MBED_ORG_USER)
 
+                # Add target's name
+                self.cxx_symbols += ["TARGET_NAME=" + self.target.name]
```

Comment:
Interesting. I thought this information would have already been available elsehwere.

Reply:
@theotherjimmy @bridadan Making sure before I press the button. Is this symbol not already available in the build system?

Reply:
Sorry a bit late to review this. To my knowledge this symbol wasn't already present. The only that was present were symbols like `TARGET_<target name>`.